appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/ExtensibleTreeMap.java

```diff
@@ -0,0 +1,48 @@
+package fish.payara.microprofile.openapi.impl.model;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.eclipse.microprofile.openapi.models.Extensible;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+
+public abstract class ExtensibleTreeMap<V, T extends Extensible<T>> extends TreeMap<String, V>
+        implements Extensible<T> {
+
+    protected Map<String, Object> extensions = new LinkedHashMap<>();
+
+    protected ExtensibleTreeMap() {
+        super();
+    }
+
+    protected ExtensibleTreeMap(Map<String, ? extends V> items) {
+        super(items);
+    }
+
+    @JsonAnyGetter
```

Comment:
If possible I'd like this in the mixin for consistency

Reply:
You mean moving the class into the `mixin` package?

Reply:
I had a look and it seemed the mixin contained all the JSON output tweaking (not just for `Extensible`) so I decided to continue this practice and added all tweaks to it (and removed them from the concrete classes).